app/services/puppet_fact_parser.rb

```diff
@@ -132,6 +132,10 @@ def support_interfaces_parsing?
     true
   end
 
+  def boot_time
+    Time.zone.now.to_i - facts[:uptime_seconds].to_i
```

Comment:
The (non-deprecated) Facter 3 fact is `system_uptime`:

```
$ facter system_uptime
{
  days => 4,
  hours => 100,
  seconds => 361049,
  uptime => "4 days"
}
```

That means you should first look for `facts[:system_uptime][:seconds]` and then fall back. I don't know if the nesting properly converts keys to symbols so probably good to check.

Reply:
updated